cmd/thanos/sidecar.go

```diff
@@ -248,6 +248,17 @@ func runSidecar(
 		g.Add(func() error {
 			defer runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
 
+			extLabelsTimeout := 2 * time.Minute
```

Comment:
Not sure if 2 min is ok, maybe worth to increase or expose via flag?

Reply:
Yeah fair point.  I've increased it, quite a lot actually to 10 minutes.  We should probably be conservative here (don't want to kill anything we shouldn't) and I'd rather avoid a flag if possible.  Feel free to adjust the value though, I might be missing some context on startup times :)